Mathlib/MeasureTheory/Group/Measure.lean

```diff
@@ -246,6 +248,27 @@ end MeasurableMul
 
 end Mul
 
+section Semigroup
+
+variable [Semigroup G] [MeasurableMul G]
+
+/-- The image of a left invariant measure under scalar multiplication is left invariant. -/
+@[to_additive "The image of a left invariant measure under scalar addition is left invariant."]
+theorem isMulLeftInvariant_map_smul
+    {α} [SMul α G] [SMulCommClass α G G] [MeasurableSpace α] [MeasurableSMul α G]
+    [IsMulLeftInvariant μ] (a : α) :
+    IsMulLeftInvariant (map (a • · : G → G) μ) :=
+  (forall_measure_preimage_mul_iff _).1 <| fun x _ hs =>
+    (SMulInvariantMeasure_map_smul μ a).measure_preimage_smul x hs
```

Comment:
Could you also add the following version and deduce that right invariant measures are preserved by left multiplication?
```
theorem isMulRightInvariant_map_smul
    {α} [SMul α G] [SMulCommClass α Gᵐᵒᵖ G] [MeasurableSpace α] [MeasurableSMul α G]
    [IsMulRightInvariant μ] (a : α) :
    IsMulRightInvariant (map (a • · : G → G) μ) :=
  (forall_measure_preimage_mul_right_iff _).1 <| fun x _ hs =>
    (SMulInvariantMeasure_map_smul μ a).measure_preimage_smul (MulOpposite.op x) hs
```

Reply:
Actually @eric-wieser I'm not sure how useful these intermediate "mul and smul" versions are: the general version is `SMulInvariantMeasure_map_smul`, and the one which are really useful in practice are the purely multiplicative one. If someone needs an intermediate case they can just use the most general version pretty easily, right?

In any case, this general version definitely allows us to prove that right invariant measures are preserved by left multiplication, so we should add that.

Reply:
Which versions are you advocating removing? I think the picture would be clearer here if `IsMulLeftInvariant` and the corresponding `SMulInvariantMeasure` were defeq to each other (though I don't know if that refactor is actually a good idea, and it certainly doesn't belong in this PR).

A point in favor of the intermediate "smul and mul" versions is that we have similar intermediate versions for algebra like `smul_mul_assoc` and `mul_smul_comm`

Reply:
What I was suggesting is going straight from `SMulInvariantMeasure_map_smul` to `isMulLeftInvariant_rmul`. But indeed the fact that there is no immediate way to go from `SMulInvariantMeasure` to `IsMulLeftInvariant` is a good enough reason to keep the intermediate step at least for now. 

I'd say we keep them as `def`s, and make the other one `instances`. Does that sound good?

Reply:
Do you mean as `theorem`s?

Reply:
I'm not sure I understood, which are you suggesting to make an `instance`?

Reply:
Using the names I suggested below: `smulInvariantMeasure_map_smul`, `isMulLeftInvariant_map_mul_right`, and `isHaarMeasure_map_mul_right`. The idea is to have instances for *smul* invariance of smul, and *mul* invariance of *mul*, does that make sense? 

Btw, I think you missed the suggestion at the beginning of this discussion :wink:

Reply:
Thanks! @ADedecker